Sources/Sequencer.swift

```diff
@@ -24,151 +24,148 @@
-      [weak self]
-      content, error in
-      // NOTE: We do not control the lifetime of the sequencer. => Fail gracefully if already released.
-      guard let this = self else { return }
+    // Cancel obsolete operations.
+    pendingOperations
+        .filter { $0.0 <= currentSeqNo - maxPendingOperationsCount }.keys
+        .forEach(cancelOperation)
 
-      // Cancel all previous requests (as this one is deemed more recent).
-      let previousRequests = this.pendingRequests.filter({ $0.0 < currentSeqNo })
-      for (seqNo, _) in previousRequests {
-        this.cancelRequest(seqNo: seqNo)
-      }
+    let operation = operationLauncher()
+    let sequencingOperation = SequencerCompletionOperation(sequenceNo: currentSeqNo, sequencer: self, correspondingOperation: operation)
+    sequencingOperation.addDependency(operation)
 
-      // Remove the current request.
-      this.pendingRequests.removeValue(forKey: currentSeqNo)
+    sequencerQueue.addOperation(sequencingOperation)
 
-      // Obsolete requests should not happen since they have been cancelled by more recent requests (see above).
-      // WARNING: Only works if the current queue is serial!
-      assert(this.lastReceivedSeqNo == nil || this.lastReceivedSeqNo! < currentSeqNo)
-
-      // Update last received response.
-      this.lastReceivedSeqNo = currentSeqNo
-
-      // Call the response handler.
-      this.delegate?.handleResponse(seqNo: currentSeqNo, content: content, error: error)
-    }
-    operation = delegate.startRequest(seqNo: currentSeqNo, completionHandler: completionHandler)
-    pendingRequests[currentSeqNo] = operation
+    pendingOperations[currentSeqNo] = operation
+    
   }
+    
+  // MARK: - Manage operations
 
-  // MARK: - Manage requests
-
-  /// Indicates whether there are any pending requests.
-  public var hasPendingRequests: Bool {
-    return !pendingRequests.isEmpty
-  }
-
-  /// Cancel all pending requests.
-  public func cancelPendingRequests() {
-    for seqNo in pendingRequests.keys {
-      cancelRequest(seqNo: seqNo)
+  /// Cancel all pending operations.
+  public func cancelPendingOperations() {
+    for seqNo in pendingOperations.keys {
+      cancelOperation(withSeqNo: seqNo)
     }
-    assert(pendingRequests.isEmpty)
+    assert(pendingOperations.isEmpty)
   }
 
-  /// Cancel a specific request.
+  /// Cancel a specific operation.
   ///
-  /// - parameter seqNo: The request's sequence number.
+  /// - parameter seqNo: The operation's sequence number.
   ///
-  public func cancelRequest(seqNo: Int) {
-    if let request = pendingRequests[seqNo] {
-      request.cancel()
-      delegate?.requestWasCancelled(seqNo: seqNo)
-      pendingRequests.removeValue(forKey: seqNo)
+  private func cancelOperation(withSeqNo seqNo: Int) {
+    if let operation = pendingOperations[seqNo] {
+      operation.cancel()
+      pendingOperations.removeValue(forKey: seqNo)
+    }
+  }
+    
+  private func dismissOperation(withSeqNo seqNo: Int) {
```

Comment:
This looks a bit similar to `cancelOperation`. Can we document the difference with `dismissOperation`?

Reply:
We actually can get rid of it as of `lastReceivedSeqNo` variable. They are used for raising an assertion in case of incorrect sequencing. Actually it's verified by unit test.

Reply:
`dismissOperation` is some kind of garbage collection after a successfully completed operation. Here we remove it from pending operations list and check if sequencing is correct by raising an assert. Should we rename to `finalizeOperation` or something like that?